projects/graph/social/social.py

```diff
@@ -45,10 +46,32 @@ def populateGraph(self, numUsers, avgFriendships):
         self.users = {}
         self.friendships = {}
         # !!!! IMPLEMENT ME
-
+        # while the number of friendships is less than the total number of users
+        if not numUsers > avgFriendships:
+            avgFriendships = numUsers - 1
         # Add users
+        for num in range(1, numUsers + 1):
+            self.addUser(num)
 
         # Create friendships
+        all_friendships = (numUsers * avgFriendships) // 2
+        friendships = []
+
+        while len(friendships) < all_friendships:
+            #generate all possible friendship combinations
+            possibilities = sorted([randint(1, numUsers), randint(1, numUsers)])
```

Comment:
this was a difficult one, but I'm hoping you were able to understand this one thoroughly, anywho goodjob!